src/lib/coerce.js

```diff
@@ -409,3 +409,53 @@ exports.crawl = function(attrs, callback, specifiedLevel) {
         if(isPlainObject(attr)) exports.crawl(attr, callback, level + 1);
     });
 };
+
+/**
+ * Find all data array attributes in a given trace object - including
+ * `arrayOk` attributes.
+ *
+ * @param {object} trace
+ *  full trace object that contains a reference to `_module.attributes`
+ *
+ * @return {array} arrayAttributes
+ *  list of array attributes for the given trace
+ */
+exports.findArrayAttributes = function(trace) {
```

Comment:
@monfera FYI I decided to make `findArrayAttributes` a lib function instead meant to be called on-demand by the transform modules. 

That way, this same function can now find the array attributes per trace module **and** filter them to match the array attribute in the user trace.